### DISCUSSION TITLE
Proposal: Null-coalescing default parameters (??=)

### DISCUSSION THREAD

**Discussion by Hyddan** [dotnet/csharplang#1826] [Language Ideas]
I gather that the null-coalescing assignment operator (??=) will be coming. I'd like to propose allowing to use that when defining default parameters.

Example syntax:
```c#
void Method(string parameter ??= "Default value") { }

Method();               // parameter would equal "Default value"
Method(null);           // parameter would equal "Default value"
Method("Other value");  // parameter would equal "Other value"
```

**Comment by bbarry** [comment]
I'd much rather prefer the breaking change of requiring the only valid default parameter to be `default`.
